### PR TITLE
CHANGELOG: add various changes introduced since v0.25

### PR DESCRIPTION
**Pull request by pks-t**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,16 +3,83 @@ v0.25 + 1
...
+    * `git_worktree_lock()` and `git_worktree_unlock()` let you lock
+      respectively unlock a worktree.
+    * `git_repository_open_from_worktree()` lets you open a repository via
+    * `git_repository_head_for_worktree()` lets you get the current `HEAD` for a
+      linked worktree.
+    * `git_repository_head_detached_for_worktree()` lets you check whether a
+      linked worktree is in detached HEAD mode.
+
+* `git_repository_item_path()` lets you retrieve paths for various repository
+  files.
```
Comment (pks-t) [inline CHANGELOG.md]: Would `git_repository_file_path()` be a better name?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,16 +3,83 @@ v0.25 + 1
...
+* `git_repository_submodule_cache_all()` and
+  `git_repository_submodule_cache_clear()` functions allow you to prime or clear
+  the submodule cache of a repository.
+
+* You can disable strict hash verifications via the
+  `GIT_OPT_ENABLE_STRICT_HASH_VERIFICATION` option with `git_libgit2_opts()`.
+
+* You can enable us calling `fsync` for various objects by setting the
+  `GIT_OPT_ENABLE_SYNCHRONIZED_OBJECT_CREATION` option with
+  `git_libgit2_opts()`.
```
Comment (pks-t) [inline CHANGELOG.md]: This option is not only about object creation, but it also modifies how we handle e.g. references. So maybe we should tweak this to `GIT_OPT_ENABLE_SYNCHRONIZED_FILE_CREATION`?

Comment (ethomson) [inline CHANGELOG.md]: Hmm.  I worry that `FILE_CREATION` makes it sound like we'll synchronize everything (including the working directory), which we don't...

`GIT_OPT_ENABLE_SYNCHRONIZED_REPO_ACCESS`...
`GIT_OPT_ENABLE_SYNCHRONIZED_REPOSITORY`...
`GIT_OPT_ENABLE_FSYNC_OBJECTS_AND_REFS`...
`GIT_OPT_ENABLE_FSYNC_REPOSITORY`...?

I dunno.

File: CHANGELOG.md (hunk 1 above)
Comment (ethomson) [inline CHANGELOG.md]: I think that I prefer `GIT_REPOSITORY_ITEM_*` and `git_repository_item_path` over `file`...  they're both a bit vague, though.  But I can't think of anything superior offhand.

I'm happy with what we have, but if you think it's _too_ vague then we can spend some brain cycles on something superior...?

Comment (pks-t) [inline CHANGELOG.md]: Let's leave it as is, then.

File: CHANGELOG.md (hunk 2 above)
Comment (pks-t) [inline CHANGELOG.md]: That's true. What about `GIT_OPT_ENABLE_FSYNC_GITDIR`? Granted, we do not currently synchronize e.g. the config files inside of the gitdir, but I think the term repo is not as unambiguous for non-git-devs as one might think.

Comment (ethomson) [inline CHANGELOG.md]: I'm 👍 on that.

Comment (ethomson) [inline CHANGELOG.md]: Do you want to send a PR to fix that up or shall I?

Comment (pks-t) [inline CHANGELOG.md]: I've got some spare time right now, will create a PR

**Comment by pks-t** [comment]
Updated to match new option name `GIT_OPT_ENABLE_FSYNC_GITDIR` introduced with #4259.

**Pull request merged**